packages/ensnode-sdk/src/ensindexer/indexing-status/types.ts

```diff
@@ -1,514 +1,741 @@
- *      - An omnichain indexing snapshot (from cache)
- *      - "now" (aka "realtime")
- * - This also includes an explicit field for `strategy` which provides
- *   more "future-proofing" and explicitly identifies how this data model is
- *   specific to omnichain indexing -- the data model and invariants would be
- *   different for other indexing strategies.
+ * Multichain indexing status snapshot when the `strategy` is
+ * {@link MultichainIndexingStrategyId.Omnichain}.
  *
  * Invariants:
- * - `realtime` is always >= `snapshot.snapshotTime`
+ * - `strategy` is always {@link MultichainIndexingStrategyId.Omnichain}.
+ * - `slowestChainIndexingCursor` is always equal to
+ *   `omnichainSnapshot.omnichainIndexingCursor`.
+ * - `snapshotTime` is always >= the "highest known block timestamp", defined as the max of:
+ *     - the `slowestChainIndexingCursor`.
+ *     - the `config.startBlock.timestamp` for all indexed chains.
+ *     - the `config.endBlock.timestamp` for all indexed chains with a `config.configType` of
+ *       {@link ChainIndexingConfigTypeIds.Definite}.
+ *     - the `backfillEndBlock.timestamp` for all chains with `chainStatus` of
+ *       {@link ChainIndexingStatusIds.Backfill}.
+ *     - the `latestKnownBlock.timestamp` for all chains with `chainStatus` of
+ *       {@link ChainIndexingStatusIds.Following}.
  */
-export type CurrentIndexingProjectionOmnichain = {
+export interface MultichainIndexingStatusSnapshotOmnichain {
   /**
-   * Indexing Strategy Type
+   * The strategy used for indexing one or more chains.
+   */
+  strategy: typeof IndexingStrategyIds.Omnichain;
+
+  /**
+   * The timestamp of the "slowest" latest indexed block timestamp
+   * across all indexed chains.
    */
+  slowestChainIndexingCursor: UnixTimestamp;
+
+  /**
+   * The timestamp when the multichain indexing status snapshot was generated.
+   *
+   * Due to possible clock skew between different systems this value must be set
+   * to the max of each of the following values to ensure all invariants are followed:
+   * - the current system time of the system generating this multichain indexing
+   *   status snapshot.
+   * - the "highest known block timestamp" (see invariants above for full definition).
+   */
+  snapshotTime: UnixTimestamp;
+
+  /**
+   * The omnichain indexing status snapshot for one or more chains.
+   */
+  omnichainSnapshot: OmnichainIndexingSnapshot;
+}
+
+/**
+ * Multichain indexing status snapshot for one or more chains.
+ *
+ * Use the `strategy` field to determine the specific type interpretation
+ * at runtime.
+ *
+ * Currently, only omnichain indexing is supported. This type could theoretically
+ * be extended to support other multichain indexing strategies in the future,
+ * such as Ponder's "multichain" indexing strategy that indexes each chain
+ * independently without deterministic ordering.
+ */
+export type MultichainIndexingStatusSnapshot = MultichainIndexingStatusSnapshotOmnichain;
```

Comment:
Ponder indexing strategies are called _mulitchain_ and _omnichain_. If we consider that ENSIndex might support the _multichain_ strategy, the type name for it would be `MultichainIndexingStatusSnapshotMultichain`. It does not read well.

Perhaps we should name the type as _crosschain_, and allow the following naming in the future:
```ts
export type CrosschainIndexingStatusSnapshot = CrosschainIndexingStatusSnapshotOmnichain | CrosschainIndexingStatusSnapshotMultichain;
```